File: SpringBootProject/src/main/java/fr/iut/sciencequest/sae/entities/scientifique/Scientifique.java
package fr.iut.sciencequest.sae.entities.scientifique;

import fr.iut.sciencequest.sae.entities.BaseEntity;
import fr.iut.sciencequest.sae.entities.Difficulte;
import fr.iut.sciencequest.sae.entities.Sexe;
import fr.iut.sciencequest.sae.entities.Thematique;
import jakarta.persistence.*;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Size;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.RequiredArgsConstructor;
import org.hibernate.validator.constraints.URL;
import org.springframework.hateoas.server.core.Relation;

import java.util.Date;

@AllArgsConstructor
@RequiredArgsConstructor
@Relation(collectionRelation = "scientifiques", itemRelation = "scientifique")
@Data
@EqualsAndHashCode(callSuper = false)
@Entity
@Table(name = "scientifique")
public class Scientifique extends BaseEntity {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;

    @ManyToOne
    @JoinColumn(name="iddifficulte")
    private Difficulte difficulte;

    @ManyToOne
    @JoinColumn(name="idthematique")
    private Thematique thematique;

    @Column(name = "photo")
    private String pathToPhoto;

    private String nom;

    private String prenom;

    private String descriptif;

    @Column(name = "datenaissance")
    private Date dateNaissance;

    @Enumerated(EnumType.STRING)
    private Sexe sexe;

    @Column(name = "ratiotrouvee")
    private double ratioTrouve = 0.0;
}

File: android/app/src/main/java/fr/iut/sciencequest/model/metier/Scientifique.kt
package fr.iut.sciencequest.model.metier

class Scientifique {
    val id: Int
    val nom: String
    val prenom: String
    val photo: String
    val descriptif: String
    val ratioTrouve: Float
    val sexe: Char
    val difficulte: Difficulte
    val thematique: Thematique

    constructor(id: Int,
                nom: String,
                prenom: String,
                photo: String,
                descriptif: String,
                ratioTrouve: Float,
                sexe: Char,
                difficulte: Difficulte,
                thematique: Thematique){
        this.id = id
        this.nom = nom
        this.prenom = prenom
        this.photo = photo
        this.descriptif = descriptif
        this.ratioTrouve = ratioTrouve
        this.sexe = sexe
        this.difficulte = difficulte
        this.thematique = thematique
    }
}